### PR TITLE
Correctly reset new token form

### PR DESCRIPTION
**Pull request by stefanv**
See also https://github.com/react-hook-form/react-hook-form/issues/1558

### CODE REVIEW

File: static/js/components/NewTokenForm.jsx [hunk 1]
```diff
@@ -24,14 +24,14 @@ import * as ProfileActions from '../ducks/profile';
...
-  const { handleSubmit, register, errors, reset, control } = useForm({
-    defaultValues
-  });
+  const { handleSubmit, register, errors, reset, control } = useForm();
+
+  useEffect(() => {
+    reset({
+      acls: Array(acls.length).fill(false)
+    });
+  }, [reset, acls]);
```
Comment (acrellin) [inline static/js/components/NewTokenForm.jsx:34]: Why is `reset` included here? Isn't that a static function? Does this work as expected without `reset` here?

Comment (acrellin) [reply static/js/components/NewTokenForm.jsx:34]: Can confirm that just `[acls]` works as expected, and I think is best here, as this hook isn't watching the value of `reset`, just `acls`.

Comment (acrellin) [reply static/js/components/NewTokenForm.jsx:34]:
```suggestion
  }, [acls]);
```

**Comment by stefanv** [comment]
I had a warning in the codebox version when removing it, but I'll investigate a bit more. I also want to take a look at useFieldArray and see if that is suitable for our needs.

**Pull request merged**